NewForce-Cohort-5/tabloidfullstack-the-pollening
# AE #23 View All User Profiles

# Description

This is for Ticket#23(https://trello.com/c/kuVg4o2D/23-23-view-all-user-profiles), which lets a user with the admin privilege see a list of all the User Profiles.

# Type of change


List of changes made

(Client-Side)
Added UserProfile route path to ApplicationViews.js, (note: UserProfileProvier was added to the app.js already)
Added category  route to Header.js 
Created UserProfile.js and UserProfileList.js
Added getAllUsers function to UserProfileProvider.js

(Server-Side)
Added method to get all categories from the sql database to the c# class in UserProfile repository and IUserProfileRepository
Updated UserProfile controller 

# Testing Instructions

**Testing objectives**
1.  To see whether the app will direct to the User Profiles list page when a user select the `User Profiles` menu option
2. To see whether the app will alphabetically display all the available UserProfiles by its display name and whether the app will also display corresponding full name and usertype when  the accordion is expanded


**Testing instructions**
(This is the similar testing instructions from the previous merged pull request.  Updated parts are **bold**.  If you have already followed these instructions below, please do **No.2**, **and go to  No.13** through No.16)

1. In Terminal, cloning down this repository to your local repository.
2. If you currently do not have access to this branch, run a <git fetch --all>, then navigate to this branch **`ae-viewAllUsers`** by typing git checkout **`ae-viewAllUsers`** in Terminal. And type Start Tabloid.sln in Terminal to run this program through Visual Studio 2022.
3. If you have not created or seeded the basic database, you will need to do so. First, open a new query by right clicking on Tabloid under the database file in the SQL Server Object Explorer on the left.
4. You will need to create the table first and then add the seed data. Choose files in the following order:
5. [01_Tabloid_Create_DB.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-the-pollening/blob/main/SQL/01_Tabloid_Create_DB.sql)
6. [02_Tabloid_Seed_Data.sql](https://github.com/NewForce-Cohort-5/tabloidfullstack-the-pollening/blob/main/SQL/02_Tabloid_Seed_Data.sql)
7. To run the query, please click the green outlined arrow under the query tab.
8. Please run the following sql commands to check the database tables have been created  and to check the data is in the   the UserProfile table

  ```
SELECT  * FROM
INFORMATION_SCHEMA.TABLES;
GO
Select * from UserProfile;
```

9. To run the back end server, please change the solid green debugger database from IISExpress to Tabloid and click the solid green debugger play button.
10. The swagger UI should run in a browser.
11. And then on the client side, please update the value of the variable called apiUrl to "https://localhost:5001/" In the UserProfile.js
12. Run `npm start` in the terminal at the level where you can see package.json
13. A browser should start to run the program with “localhost” as a part of the URL. The opening page should be the Log-in page. Please log in using this email address that should be in your database to log in: [foo@bar.com](mailto:foo@bar.com)
14. Please select UserProfiles in the header and to see if the app direct you to the User Profiles list page (Testing objective1)
15. Please check to see if the list of user profiles are displayed alphabetically by display name when you select the User Profile on the header and expand the accordion to see if the app will display full name and user type by name. (Testing objective 2)
16. If there are no issues, please go back to Terminal and run the following command ‘git checkout main’ to get back to the local main branch



# Checklist:
